opp_path_selection/include/opp_path_selection/filter.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OPP_PATH_SELECTION_FILTER_H
+#define OPP_PATH_SELECTION_FILTER_H
+
+#include <pcl/point_cloud.h>
+
+namespace opp_path_selection
+{
+namespace data_filtering
+{
+template <typename PointT>
+using Cloud = typename pcl::PointCloud<PointT>;
+
+template <typename PointT>
+using CloudPtr = typename Cloud<PointT>::Ptr;
+
+template <typename PointT>
+bool planeFit(const CloudPtr<PointT> input_cloud,
+              Cloud<PointT>& output_cloud,
+              pcl::ModelCoefficients::Ptr plane_coefficients,
+              const double threshold = 0.025);
+
+}  // end namespace data_filtering
+
+}  // end namespace opp_path_selection
+
+#include "opp_path_selection/filter_impl.h"
```

Comment:
I would suggest not including the implementation in the header. Rather, build a library with specific implementations of the template (using the `PCL_INSTANTIATE` macro). If you need a type that isn't already built, I would prefer to include the implementation directly.

Reply:
This is a direct copy of the area_selector code with the changes necessary to create a path rather than an area. I would have never ever created the xxx_impl.h otherwise. I do not understand what you are suggesting. I'll take this one off line.